#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/OsbWood.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"

WorldInfo {
  info [
    "In this example, the robot moves down a slope."
    "The altimeter is used to record the robots height."
  ]
  title "Altimeter"
  basicTimeStep 8
  contactProperties [
    ContactProperties {
      material2 "body"
      coulombFriction [
        0
      ]
      softCFM 0.1
    }
  ]
}
Viewpoint {
  orientation -0.04249299771615962 0.03597323812395563 0.998448932737159 5.024399358080451
  position -0.5099567495608207 1.3747014201881598 0.3091119993044244
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
}
PointLight {
  attenuation 0 0 1
  intensity 0.6
  location 0.6 0 0
}
DEF SLOPE Solid {
  translation 0 0 0.15
  rotation 0 1 0 -0.4
  children [
    Shape {
      appearance OsbWood {
        textureTransform TextureTransform {
          scale 0.5 2
        }
      }
      geometry DEF BOX0 Box {
        size 0.8 0.2 0.02
      }
    }
  ]
  name "green box"
  boundingObject USE BOX0
}
Robot {
  translation -0.11280420413782008 0.0021379365276690747 0.11326923370772614
  rotation -0.24579639186324329 -0.003192930350732898 0.9693162223664609 3.09987416359558
  children [
    Altimeter {
    }
    Pose {
      translation 0 0 0.0415
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 3069.352638400881
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 0.044999999999999825 0.025
        rotation -0.5805214984609255 0.5805215665324576 -0.5709549024357203 2.104014718762067
        children [
          DEF WHEEL Pose {
            rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
        linearVelocity 0.04600270555434038 -0.0021095571439872472 0.019449739793862397
        angularVelocity 0.07770854137496429 1.998219063270983 0.032887587487012905
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 3069.3526385320242
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 -0.045 0.025
        rotation -0.5805215232824088 0.580521591353939 -0.5709548519609603 2.1040147948935393
        children [
          USE WHEEL
        ]
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
        linearVelocity 0.046001757804297855 -0.002109510121451014 0.0194491154035491
        angularVelocity 0.07770859326243836 1.9982190271806162 0.03288764511913254
      }
    }
    DistanceSensor {
      translation 0.042 0.02 0.063
      rotation 0 0 -1 -0.499205
      children [
        DEF SENSOR Pose {
          rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.042 -0.02 0.063
      rotation 0 0 1 -0.500795
      children [
        USE SENSOR
      ]
      name "ds1"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DEF SMILE Pose {
      translation 0.036 0 0.025
      rotation 0.25056300542381216 -0.9351130202419243 -0.25056300542381216 1.637834
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.290196 0.290196
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
  ]
  name "MyBot"
  contactMaterial "body"
  boundingObject Pose {
    translation 0 0 0.0415
    rotation 0 0 1 -1.570796
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "Class-282"
  linearVelocity 0.04600223498340407 -0.002109515266230232 0.01944942930759338
  angularVelocity -6.98797939254437e-06 2.768122915354245e-07 1.0559090179841235e-05
}
